Move THTensor_compute_stride to ATen

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23121 [quant] Verify flatten works for quantized Tensor
* #23046 support at::view and at::reshape for quantized tensor
* **#23045 Move THTensor_compute_stride to ATen**

att

Differential Revision: [D16368860](https://our.internmc.facebook.com/intern/diff/D16368860/)

jerryzh168 committed:
Move THTensor_compute_stride to ATen

att

Differential Revision: [D16368860](https://our.internmc.facebook.com/intern/diff/D16368860/)

dzhulgakov commented on aten/src/ATen/TensorUtils.cpp:
```diff
@@ -311,5 +311,73 @@ int64_t computeStorageSize(IntArrayRef sizes, IntArrayRef strides) {
+//    where each chunk of newshape has matching ``numel'', i.e., number of subspaces,
+//    as the corresponding chunk of oldshape.
+c10::optional<std::vector<int64_t>> computeStride(
+    at::IntArrayRef oldshape,
```
nit: you don't even need at:: anymore

jerryzh168 committed:
Update on "Move THTensor_compute_stride to ATen"


att

Differential Revision: [D16368860](https://our.internmc.facebook.com/intern/diff/D16368860/)
